collections/mod.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+
+/*
+ * Stack is a first-in-last-out data structure.
+ *
+ * Create by optionally passing an Iterable:
+ *
+ *      const s = Stack([1, 2, 3])
+ *      console.log(s.pop())  // 3
+ *
+ */
+export class Stack<T> implements Iterable<T> {
+  private _array: T[];
+  constructor(ls?: Iterable<T>) {
+    this.clear();
+    for (const item of ls || []) {
+      this._array.push(item);
+    }
+  }
+  clear(): void {
+    this._array = [];
+  }
+  peek(): T {
+    return this._array[this._array.length - 1];
+  }
+  pop(): T {
+    return this._array.pop();
+  }
+  push(value): void {
+    this._array.push(value);
+  }
+  size(): number {
+    return this._array.length;
+  }
+
+  [Symbol.iterator]() {
+    const arr = this._array;
+    let index = arr.length;
+    return {
+      next: function() {
+        index--;
+        return {
+          done: index < 0,
+          value: arr[index]
+        };
+      }
+    };
+  }
+}
+
+/*
```

Comment:
can you split this into two files `//collections/stack.ts` and `//collections/queue.ts` with corresponding test files. No `mod.ts` nor `test.ts`.

Reply:
So the convention is every class will be in its own file and need its own import line? 😕

Reply:
I'm not a fan on `mod.ts` it's better to be descriptive...

Reply:
The annoying part of that is:

```ts
import { Deque, Stack, Queue } from ".../std/collections/mod.ts";
```
becomes (for every function/class etc)
```ts
import { Deque } from ".../std/collections/deque.ts";
import { Stack } from ".../std/collections/stack.ts";
import { Queue } from ".../std/collections/queue.ts";
```

it is convenient to group functions/classes together. The function/class name is the thing that's descriptive.

Reply:
Done, but for the record: I really don't like it.